contracts/pToken.sol

```diff
@@ -78,13 +80,15 @@ contract PToken is
         );
         _mint(recipient, value, userData, operatorData);
         if (userData.length > 0) {
-            try IPReceiver(recipient).receiveUserData(userData) {} catch {
-                // The recipient may be not implementing the IPReceiver interface,
-                // or the receiveUserData() hook may have a bug inside that would
-                // revert the entire transaction, thus blocking the minting process.
-                // Swallowing the error here permits pNetwork to fulfill its role
-                // of relaying messages/tokens across chains.
-            }
+            // pNetwork aims to deliver cross chain messages successfully regardless of what the user may do with them.
+            // We do not want this mint transaction reverting if their receiveUserData function reverts,
+            // and thus we swallow any such errors, emitting a `ReceiveUserDataFailed` event instead.
+            // The low-level call is used because in the solidity version this contract was written in,
+            // a try/catch block fails to catch the revert caused if the receiver is not in fact a contract.
+            // This way, a user also has the option include userData even when minting to an externally owned account.
```

Comment:
This comment needs updating since we're not longer using the `try/catch`.

Reply:
Comment updated